cpp/src/core/structures/trajectory.h

```diff
@@ -213,7 +214,9 @@ class Trajectory {
     }
     double insertion_duration_cost(size_t insert_loc, const Segment3d segment) const {
         auto increase_in_length =  insertion_length_cost(insert_loc, segment);
-        ASSERT(ALMOST_GREATER_EQUAL(increase_in_length, 0))
+        if (!ALMOST_GREATER_EQUAL(increase_in_length, 0)) {
+            std::cerr << "WARNING: increase in length < 0 : " << increase_in_length << std::endl;
```

Comment:
In general, I'd rather have an assert because it forces us to handle the problem when it shows up and it makes the problem appear in automated tests or large runs.

Here failure to fullfill the test would not show up in continuous integration. If the assert is to strict, it might be relaxed to make sure reasonable results are accepted.

Reply:
You are right. I dropped the commit.